### DISCUSSION TITLE
Volume mapped to contrast for my Display?

### DISCUSSION THREAD

**Discussion by robcalcroft** [MonitorControl/MonitorControl#797] [Q&A]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Volume doesn't work and appears in some cases to be mapped to contrast instead, i have tried the https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting#try-a-different-ddc-tool tool and brightness works but volume doesn't. There are no errors shown and the app appears to think it can control volume and the keys respond correctly but i dont hear a change in volume.

### Expected behavior

Volume should change on pressing hardware buttons or changing slider in MonitorControl interface

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.6.1
- Mac model: MacBook Pro (16-inch, 2019)
- MonitorControl version: 4.0.1
- Monitor(s): MSI Optix MAG342CQRV
- Monitor Cable(s)/Connection(s): USB-C to https://www.amazon.co.uk/gp/product/B08H5CYC43/ref=ppx_yo_dt_b_asin_title_o01_s01?ie=UTF8&psc=1 to Monitor
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
This seems to be a monitor specific issue. It might be that the display's firmware incorrectly maps the volume command to brightness.

I'd advise you to go to Displays, enable `Show advanced settings` and try changing the VCP value for the volume command to something that the display accepts as volume control code.

You can see the control codes according to the standard here but not all displays follow the specification:

https://github.com/MonitorControl/MonitorControl/blob/master/MonitorControl/Enums/Command.swift

so for example volume is set by `audioSpeakerVolume` which has a control code of `62` by default. Try using a different number from the list until you find one that might work (you might want to try `contrast` which is `12`, maybe these are reversed in the firmware).

<img width="842" alt="Screen Shot 2021-11-18 at 19 20 37" src="https://user-images.githubusercontent.com/37590873/142473925-7e896f33-dd12-4d02-92c5-85a754d118a5.png">

**Reply by robcalcroft** [reply]
Hmm, i tried all the volume related values and none of them worked :/ im starting to think this monitor might not support volume change, it doesnt have a volume control in the options on the monitor itself so i think im out of luck.
Do you know of a way i can do software volume control? Like how the software brightness can be done?

**Reply by waydabber** [reply]
Hi, yes probably the monitor simply gives a line out and that's all. You can use apps like [eQMac](https://eqmac.app) for software control.
